codegenerator/cli/templates/static/codegen/src/db/Migrations.res

```diff
@@ -70,12 +73,15 @@ let createEnumIfNotExists = (sql, enum: Enum.enum<_>) => {
   open Belt
   let {variants, name} = enum
   let mappedVariants = variants->Array.map(v => `'${v->Utils.magic}'`)->Js.Array2.joinWith(", ")
-  let query = `
-      DO $$ BEGIN
-      IF NOT EXISTS(SELECT 1 FROM pg_type WHERE typname = '${name->Js.String2.toLowerCase}') THEN
-        CREATE TYPE ${name} AS ENUM(${mappedVariants});
-        END IF;
-      END $$; `
+  let query = `DO $$ BEGIN
+  IF NOT EXISTS (
+    SELECT 1 
+    FROM pg_type 
+    WHERE typname = '${name->Js.String2.toLowerCase}' 
+    AND typnamespace = (SELECT oid FROM pg_namespace WHERE nspname = '${Env.Db.publicSchema}')
```

Comment:
Wow, devils here hey, was this caught by a test? :)

Reply:
Yep